.orchestra/ci/ci-hook/app.py

```diff
@@ -132,34 +186,206 @@ def trigger_ci(username, repo_url, ref, before, after):
-               data["ref"],
-               data["before"],
-               data["after"])
-    return "All good\n"
+    attributes = data.get("object_attributes", {})
+
+    if event == "Push Hook":
+        trigger_ci(
+            data["user_username"],
+            data["project"]["git_http_url"] + " " + data["project"]["git_ssh_url"],
+            data["ref"]
+        )
+
+    elif event == "Merge Request Hook" and attributes.get("action", "") in ("open", "update") \
+            and attributes.get("state", "") == "opened":
+        trigger_ci(
+            data["user"]["username"],
+            attributes["source"]["git_http_url"] + " " + attributes["source"]["git_ssh_url"],
+            attributes["source_branch"],
+            {
+                "platform": "gitlab",
+                "gitlab_project_id": data['project']['id'],
+                "gitlab_mr_iid": attributes['iid'],
+                "head_sha": attributes['last_commit']['id']
+            }
+        )
+
+    return "All good\n", 200
 
 
 @app.route('/ci-hook/github', methods=["POST"])
 def github_hook():
     headers = dict(request.headers)
-    signature = 'sha256=' + hmac.new(GITHUB_SECRET, request.data, hashlib.sha256).hexdigest()
+
+    event = headers.get("X-Github-Event", "")
+    if event in ("check_suite", "check_run"):
+        secret = GITHUB_APP_SECRET
+    else:
+        secret = GITHUB_SECRET
+
+    signature = 'sha256=' + hmac.new(secret, request.data, hashlib.sha256).hexdigest()
     if signature != headers.get("X-Hub-Signature-256"):
         return "Invalid signature\n", 403
 
-    if headers.get("X-Github-Event", "") == "push":
-        data = request.json
-        trigger_ci(hub_to_lab(data["sender"]["login"]),
-                   data["repository"]["clone_url"],
-                   data["ref"],
-                   data["before"],
-                   data["after"])
+    event = headers.get("X-Github-Event", "")
+    data = request.json
 
-    return "All good\n"
+    if event == "push":
+        trigger_ci(
+            hub_to_lab(data["sender"]["login"]),
+            data["repository"]["clone_url"],
+            data["ref"]
+        )
+    elif event in ("check_suite", "check_run") and data.get("action", "") in ('requested', 'rerequested'):
+        # GitHub is requesting a check run; we have to first confirm it, then when we receive the `created` action we
+        # can actually run the CI
+        r = requests.post(
+            f"{GITHUB_API_URL}/repos/{data['repository']['full_name']}/check-runs",
+            headers=github_headers(),
+            json={
+                "name": "rev.ng CI",
+                "head_sha": 'check_run' in data and data['check_run']['head_sha'] or data['check_suite']['head_sha']
+            }
+        )
+        if r.status_code not in (200, 201, 202):
+            return f"Check run creation failed:\n{r.content.decode(errors='replace')}\n", r.status_code
```

Comment:
Could make sense, but it is also useful to return it as a response since you can visit the GitHub app page, view failed webhooks and see the response.